File: SeongYeon/baekjoon/10799.js
// input
const input = require('fs').readFileSync('test/input2.txt').toString().trim();

// solution
let result = 0;
const stack = [];
for (let i = 0; i < input.length; i++) {
  if (input[i] === '(') stack.push(0);
  else {
    stack.pop();
    if (input[i - 1] === '(') result += stack.length;
    else result += 1;
  }
}

// output
console.log(result);

File: JiHwan/BJ/10799.py
razor = list(input())
result = 0
stack = []

for i in range(len(razor)):
    if razor[i] == '(':
        stack.append(razor[i])

    else :
        if razor[i-1] == '(':
            stack.pop()
            result +=len(stack)
        else:
            stack.pop()
            result += 1
print(result)